### PR TITLE
FIXED THE DATABASE ERRORS

### PR DESCRIPTION
**Pull request by PajakKonrad**
Fixed the database and data binding errors.

### CODE REVIEW

File: Data/Database.cs [hunk 1]
```diff
@@ -10,14 +10,42 @@ namespace MauiApp1.Data
...
 
-        async Task Init()
+        public async Task InitializeDatabaseAsync()
         {
             if (_database is not null)
                 return;
 
-            _database = new SQLiteAsyncConnection(DatabaseSettings.DBPath, DatabaseSettings.Flags);
-            // TODO: Create tables that we will use for project
+            try
```
Comment (byVirrox) [inline Data/Database.cs:23]: Remove comments

**Review by byVirrox — APPROVED** [review]

File: Data/Database.cs (hunk 1 above)
Comment (PajakKonrad) [reply Data/Database.cs:23]: this wasnt my comment it was there when I started fixing it and its red so its gone :P

File: Data/Database.cs [hunk 2]
```diff
@@ -10,14 +10,52 @@ namespace MauiApp1.Data
     public class Database
     {
         public SQLiteAsyncConnection _database;
+        public Database()
+        {
+            InitializeDatabaseAsync().Wait(); // Ensure initialization is complete before proceeding
```
Comment (Pannoniae) [inline Data/Database.cs:15]: Use .Result instead of Wait(), this swallows the exception if InitializeDatabaseAsync throws one.

File: Data/Database.cs [hunk 3]
```diff
@@ -10,14 +10,52 @@ namespace MauiApp1.Data
...
+            {
+                // Create your database tables here
+                await _database.CreateTableAsync<Models.AlertType>();
+                await _database.CreateTableAsync<Models.OperationalTeamStatus>();
+                await _database.CreateTableAsync<Models.BuildingType>();
+                await _database.CreateTableAsync<Models.Continent>();
+                await _database.CreateTableAsync<Models.Equipment>();
+                await _database.CreateTableAsync<Models.OrderStatus>();
+                await _database.CreateTableAsync<Models.Organisation>();
+                await _database.CreateTableAsync<Models.position_statuses>();
```
Comment (Pannoniae) [inline Data/Database.cs:46]: this should be renamed properly, maybe to PositionStatus?

File: Models/BuildingType.cs [hunk 4]
```diff
@@ -22,36 +22,27 @@ public class BuildingType : INotifyPropertyChanged
...
 
-        private string name;
-
-        /// <summary>
-        /// The name of the building type.
-        /// </summary>
-        public string Name {
-            get => name;
-            set => SetField(ref name, value);
+        private string _name;
```
Comment (Pannoniae) [inline Models/BuildingType.cs:25]: why did you remove all the comments

File: Models/BuildingType.cs [hunk 5]
```diff
@@ -22,36 +22,27 @@ public class BuildingType : INotifyPropertyChanged
...
 
-        /// <summary>
-        /// Reflection helper for implementing <see cref="INotifyPropertyChanged"/>.
-        /// </summary>
-        /// <typeparam name="T">The type of the field.</typeparam>
-        /// <param name="field">A reference to the field.</param>
-        /// <param name="value">The value of the field.</param>
-        /// <param name="propertyName">The property name changing. This is automatically assigned by <see cref="CallerMemberNameAttribute"/>.</param>
-        /// <returns></returns>
+        protected bool SetProperty<T>(ref T storage, T value, [CallerMemberName] string propertyName = null)
```
Comment (Pannoniae) [inline Models/BuildingType.cs:38]: why did you remove all the comments, they are useful, this is a hairy method

File: Views/PositionStatusPage.xaml.cs [hunk 6]
```diff
@@ -21,7 +21,7 @@ partial class position_status : ContentPage
     public position_status()
```
Comment (Pannoniae) [inline Views/PositionStatusPage.xaml.cs:21]: This should be renamed as well

**Review by Pannoniae — CHANGES_REQUESTED** [review]
Looks nice. Please put the comments back, they were useful.

File: Data/Database.cs (hunk 3 above)
Comment (PajakKonrad) [reply Data/Database.cs:46]: Yeah for some reason actual name in solution explorer differs the actual name , i just worked with what I've got :)

File: Models/BuildingType.cs (hunk 4 above)
Comment (PajakKonrad) [reply Models/BuildingType.cs:25]: Because I copy pasted the bit of code which was working and added it to every class all of them have same comments.
